From: "Barak A. Pearlmutter" <bap@debian.org>
Date: Sun, 24 Feb 2019 16:49:20 +0000
Subject: cast warnings

---
 src/ttsubset/sft.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/ttsubset/sft.c b/src/ttsubset/sft.c
index 249fbdf..619753f 100644
--- a/src/ttsubset/sft.c
+++ b/src/ttsubset/sft.c
@@ -739,7 +739,7 @@ static int GetCompoundTTOutline(TrueTypeFont *ttf, guint32 glyphID, ControlPoint
         index = GetUInt16(ptr, 2, 1);
         ptr += 4;
 
-        if (listFind(glyphlist, (void *) (int) index)) {
+        if (listFind(glyphlist, (void *) (long) index)) {
 #ifdef DEBUG
             fprintf(stderr, "Endless loop found in a compound glyph.\n");
             fprintf(stderr, "%d -> ", index);
@@ -753,7 +753,7 @@ static int GetCompoundTTOutline(TrueTypeFont *ttf, guint32 glyphID, ControlPoint
 #endif
         }
 
-        listAppend(glyphlist, (void *) (int) index);
+        listAppend(glyphlist, (void *) (long) index);
 
 #ifdef DEBUG2
         fprintf(stderr,"glyphlist: += %d\n", index);
@@ -937,7 +937,7 @@ static int GetTTGlyphOutline(TrueTypeFont *ttf, guint32 glyphID, ControlPoint **
         if (!glyphlist) {
             glyphlistFlag = 1;
             glyphlist = listNewEmpty();
-            listAppend(glyphlist, (void *) glyphID);
+            listAppend(glyphlist, (void *) (long) glyphID);
         }
         res = GetCompoundTTOutline(ttf, glyphID, pointArray, metrics, glyphlist);
         if (glyphlistFlag) {
@@ -1844,7 +1844,7 @@ int GetTTGlyphComponents(TrueTypeFont *ttf, guint32 glyphID, list glyphlist)
     if (glyphID >= ttf->nglyphs) return 0;
     ptr = glyf + ttf->goffsets[glyphID];
 
-    listAppend(glyphlist, (void *) glyphID);
+    listAppend(glyphlist, (void *) (long) glyphID);
 
     if (GetInt16(ptr, 0, 1) == -1) {
         guint16 flags, index;
