From: "Barak A. Pearlmutter" <bap@debian.org>
Date: Sun, 24 Feb 2019 22:32:27 +0000
Subject: format warnings

---
 src/xo-print.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/xo-print.c b/src/xo-print.c
index 3be1843..dfba40d 100644
--- a/src/xo-print.c
+++ b/src/xo-print.c
@@ -821,10 +821,10 @@ gboolean pdf_draw_image(PdfImage *image, struct XrefTable *xref, GString *pdfbuf
 
   xref->data[image->n_obj] = pdfbuf->len;
   g_string_append_printf(pdfbuf, 
-    "%d 0 obj\n<< /Length %d /Filter /FlateDecode /Type /Xobject "
+    "%d 0 obj\n<< /Length %lu /Filter /FlateDecode /Type /Xobject "
     "/Subtype /Image /Width %d /Height %d /ColorSpace /DeviceRGB "
     "/BitsPerComponent 8 ",
-    image->n_obj, zpix->len, width, height);
+    image->n_obj, (unsigned long) zpix->len, width, height);
   if (image->has_alpha) {
     g_string_append_printf(pdfbuf, 
       "/SMask %d 0 R ",
@@ -850,10 +850,10 @@ gboolean pdf_draw_image(PdfImage *image, struct XrefTable *xref, GString *pdfbuf
     
     xref->data[image->n_obj_smask] = pdfbuf->len;
     g_string_append_printf(pdfbuf, 
-      "%d 0 obj\n<< /Length %d /Filter /FlateDecode /Type /Xobject "
+      "%d 0 obj\n<< /Length %lu /Filter /FlateDecode /Type /Xobject "
       "/Subtype /Image /Width %d /Height %d /ColorSpace /DeviceGray "
       "/BitsPerComponent 8 >> stream\n",
-      image->n_obj_smask, zpix->len, width, height);
+      image->n_obj_smask, (unsigned long) zpix->len, width, height);
 
     g_string_append_len(pdfbuf, zpix->str, zpix->len);
     g_string_free(zpix, TRUE);
