From: Denis Auroux <auroux@users.sourceforge.net>
Date: Sun, 24 Jan 2021 16:15:18 -0500
Subject: control+mousewheel scrolls in/out (only if XInput option enabled)
 (2nd attempt)

---
 src/xo-callbacks.c | 28 +++++++++++++++++-----------
 1 file changed, 17 insertions(+), 11 deletions(-)

diff --git a/src/xo-callbacks.c b/src/xo-callbacks.c
index 568bdde..b4e6339 100644
--- a/src/xo-callbacks.c
+++ b/src/xo-callbacks.c
@@ -2423,6 +2423,19 @@ on_canvas_button_press_event           (GtkWidget       *widget,
          so the core button event gets discarded and the scroll event never 
          gets processed by the main window. This is arguably a GTK+ bug.
          We work around it. */
+
+      // control-mousewheel should zoom; standard zoom step factor is too steep
+      if (event->state & GDK_CONTROL_MASK) {
+        if (event->button == 4 && ui.zoom < MAX_ZOOM) ui.zoom *= 1.1;
+        if (event->button == 5 && ui.zoom > MIN_ZOOM) ui.zoom /= 1.1;
+        gnome_canvas_set_pixels_per_unit(canvas, ui.zoom);
+        rescale_text_items();
+        rescale_bg_pixmaps();
+        rescale_images();
+        return TRUE; // event handled, stop
+      }
+      
+      // convert things into a scroll event
       scroll_event.scroll.type = GDK_SCROLL;
       scroll_event.scroll.window = event->window;
       scroll_event.scroll.send_event = event->send_event;
@@ -2433,21 +2446,14 @@ on_canvas_button_press_event           (GtkWidget       *widget,
       scroll_event.scroll.device = event->device;
       scroll_event.scroll.x_root = event->x_root;
       scroll_event.scroll.y_root = event->y_root;
-      // scroll direction
+
+      // scroll direction; shift+mousewheel should do horizontal scroll
+      if (event->button < 6 && event->state & GDK_SHIFT_MASK) event->button += 2;
       if (event->button == 4) scroll_event.scroll.direction = GDK_SCROLL_UP;
       else if (event->button == 5) scroll_event.scroll.direction = GDK_SCROLL_DOWN;
       else if (event->button == 6) scroll_event.scroll.direction = GDK_SCROLL_LEFT;
       else scroll_event.scroll.direction = GDK_SCROLL_RIGHT;
-      // control-mousewheel should zoom; standard zoom step factor is too steep
-      if (event->state & GDK_CONTROL_MASK) {
-        if (event->button == 4 && ui.zoom < MAX_ZOOM) ui.zoom *= 1.1;
-        if (event->button == 5 && ui.zoom > MIN_ZOOM) ui.zoom /= 1.1;
-        gnome_canvas_set_pixels_per_unit(canvas, ui.zoom);
-        rescale_text_items();
-        rescale_bg_pixmaps();
-        rescale_images();
-        return FALSE;
-      }
+
       gtk_widget_event(GET_COMPONENT("scrolledwindowMain"), &scroll_event);
     }
     return FALSE;
